contrib/pg_tde/src/pg_tde_event_capture.c

```diff
@@ -273,35 +273,24 @@ pg_tde_ddl_command_end_capture(PG_FUNCTION_ARGS)
 	if (IsA(parsetree, AlterTableStmt) && tdeCurrentCreateEvent.alterAccessMethodMode)
 	{
 		/*
-		 * sequences are not updated automatically call a helper function that
-		 * automatically alters all of them, forcing an update on the
-		 * encryption status int ret;
+		 * sequences are not updated automatically so force rewrite by
+		 * updating their presistence to be the same as before.
 		 */
-		char	   *sql = "SELECT pg_tde_internal_refresh_sequences($1);";
-		Oid			argtypes[1];
-		SPIPlanPtr	plan;
-		Datum		args[1];
-		char		nulls[1];
-		int			ret;
+		List	   *seqlist = getOwnedSequences(tdeCurrentCreateEvent.baseTableOid);
+		ListCell   *lc;
+		Relation	rel = relation_open(tdeCurrentCreateEvent.baseTableOid, NoLock);
+		char		persistence = rel->rd_rel->relpersistence;
 
-		SPI_connect();
+		relation_close(rel, NoLock);
```

Comment:
Shouldn't we keep some level of lock, or add error handling around this code?

In case somebody does a parallel alter sequence, and changes the ownership of one of the sequences?

Reply:
The relation will be locked exclusively since we changed access method  so that is not an issue. I could add an assert in a branch to verify that though so I am not confused.

Reply:
Yes, I just asked because I'm not 100% sure about it. It makes sense that the after ddl event trigger is executed in the same transaction and we still have all the locks, but I realized that I never verified this when I started using this function to do actual changes.

Reply:
>Note that the trigger fires after the actions have taken place (but before the transaction commits), and thus the system catalogs can be read as already changed.

> Similarly, if a ddl_command_end trigger fails with an error, the effects of the DDL statement will be rolled back, just as they would be in any other case where the containing transaction aborts.

Okay, this shouldn't be an issue, I just had to read the documentation.

Reply:
Great!